### PR TITLE
wireless/wapi: add save_config/reconnect/scan_results support

### PR DESCRIPTION
**Pull request by anchao**

### CODE REVIEW

**Comment by anchao** [comment]
so, i think we can ignore the warning here like:
...
wireless/wapi/src/util.c:79:6: error: Mixed case identifier found
wireless/wapi/src/util.c:106:9: error: Mixed case identifier found
wireless/wapi/src/util.c:122:33: error: Mixed case identifier found
wireless/wapi/src/util.c:128:6: error: Mixed case identifier found
...
since the cJson api is mixed case.

File: netutils/netinit/netinit_associate.c [hunk 1]
```diff
@@ -77,7 +77,8 @@ int netinit_associate(FAR const char *ifname)
       conf.phraselen   = strlen(conf.passphrase);
     }
 
-  ret = wpa_driver_wext_associate(&conf);
+  if (conf.ssidlen > 0)
+    ret = wpa_driver_wext_associate(&conf);
```
Comment (Ouss4) [inline netutils/netinit/netinit_associate.c:81]:
```suggestion
  if (conf.ssidlen > 0)
    {
      ret = wpa_driver_wext_associate(&conf);
    }
```

File: netutils/netinit/netinit_associate.c [hunk 2]
```diff
@@ -60,7 +60,7 @@
 int netinit_associate(FAR const char *ifname)
 {
   struct wpa_wconfig_s conf;
-  int ret;
+  int ret = -EINVAL;
```
Comment (Ouss4) [inline netutils/netinit/netinit_associate.c:63]: This is causing the build to fail. errno.h is not included.

File: netutils/netinit/netinit_associate.c (hunk 1 above)
Comment (anchao) [reply netutils/netinit/netinit_associate.c:81]: Oops, Done

File: netutils/netinit/netinit_associate.c (hunk 2 above)
Comment (anchao) [reply netutils/netinit/netinit_associate.c:63]: Done

**Pull request merged**